### PR TITLE
Re #14883: Reimplemented QECoverage as an interface.

### PR DESCRIPTION
**Pull request by mducle**
Reimplementation of PR #14884 as a GUI interface rather than algorithm, addressing issue #14883, because I couldn't figure out how to revert git changes.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
Where is the documentation for this new interface?

**Comment by mducle** [comment]
https://github.com/mducle/mantid/blob/mducle/14883_qecoverage_interface/docs/source/interfaces/QECoverage.rst

**Comment by KarlPalmen** [comment]
Thank you for the information. Sorry I didn't see it in your first commit straight away.

**Comment by mducle** [comment]
No probs... my first contribution to Mantid, and I'm still learning how to use github. 

It might be good in future to read the instrument definition xml files to determine the two theta limits, but for now I think this works, and it would take more time than I want to spend to do that.

File: docs/source/interfaces/QECoverage.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+For direct geometry spectrometers, an incident energy (Ei) in meV must be given; for indirect geometry machines,
+the final energies are fixed at a few values which must be selected from the drop-down menu.
+
+To overplot the calculated Q-E trajectories, set the "Plot Over" check box.
+
+You can choose to create a 1D Mantid workspace for latter plotting using the "Create Workspace" check box.
+
+In addition, for HYSPEC, where the detector bank can be rotated, an additional parameter **S2** denoting the
+scattering angle of the center of the detector bank is required. The detectors extend over 60 degrees, so
+the two theta limits will be taken from min([0, abs(s2)-30]) to abs(s2)+30.
```
Comment (AndreiSavici) [inline docs/source/interfaces/QECoverage.rst]: Replace min with max. If s2 is 10, tt goes from 0 to 40 (really it is -20 to 40, but for powder 0->20 and 0->-20 are equivalent)

**Comment by mducle** [comment]
Oops. Yes. You're right...

**Comment by KarlPalmen** [comment]
It would be useful to describe what one needs to do to test it, like in most other pull requests.
Else I would only be able verify, the interface exists and is like how it described in the documentation.

**Comment by mducle** [comment]
Ah right... Well, it's actually a pretty simple calculation, and most of the code is in the gui itself. 

You can test the gui functionality:

1. Selecting Hyspec should show a edit box labelled "S2", and changing the value of S2 should shift the plotted curves left (decreasing S2) or right (increasing S2).
2. Checking either of the plot over or create workspace check boxes on one of the Direct or Indirect tabs should check them in the other tab.
3. Plot over functionality works (matplotlib 'hold' should be set) so you can overplot curves.

As for what it should look like, the curves for Maps and Sequoia should look similar, whilst that for all the other direct geometry instruments should be similar for a given Ei. The curve for Vesuvio should extend to the 1000's of meV range, whilst everything else is in the 1-10's of meV range.

Sorry, if this is not very specific. 

Since this is meant to help experiment planing, do you think it would be better to be in a tab in the DGSPlanner interface?

**Comment by KarlPalmen** [comment]
I don't know enough about experiment planning to comment about whether it would be better as a tab in the DGSPl;anner interface. Perhaps someone else will comment.

**Comment by AndreiSavici** [comment]
Needs release notes

**Comment by mducle** [comment]
Thanks!

I've just asked for a mantidproject.org account, so I'll add the release notes to the page when this gets through.

**Pull request merged**